### PR TITLE
simplify error conversion

### PR DESCRIPTION
**Pull request by conorpp**
Did this while figuring out what ended up being an unrelated error.  I think it's a bit simpler and takes up ~600 less bytes in program space.

### CODE REVIEW

File: src/fs.rs [hunk 1]
```diff
@@ -951,7 +951,7 @@ impl<S: driver::Storage> io::Read for File<'_, '_, S>
             buf.as_mut_ptr() as *mut cty::c_void,
             buf.len() as u32,
         ) };
-        io::result_from(return_code).map(|read| read as usize)
+        io::result_from_error(return_code)
```
Comment (nickray) [inline src/fs.rs:954]: This should be `io::result_from(read as usize, return_code)` I think? (And the following two with position/written).

**Review by nickray — CHANGES_REQUESTED** [review]

File: src/fs.rs (hunk 1 above)
Comment (conorpp) [reply src/fs.rs:954]: My thought was that casting it to usize before checking if it's negative was a bit weird, but it works in any event.

**Comment by conorpp** [comment]
@nickray addressed comment

**Comment by nickray** [comment]
Maybe I misunderstood, I thought the read/write methods were no longer returning the number of bytes they actually read/wrote. Was this already the case before addressing? And/or is it now? If I remember correctly, one cannot assume that everything will be read/written in one go.

**Comment by conorpp** [comment]
I didn't change anything functionally, just refactored.  All the return values are the same, so yes still being returned and was the case before.

**Pull request merged**